cmplibrary/src/test/java/com/sourcepoint/cmplibrary/SourcePointClientTest.java

```diff
@@ -77,129 +75,80 @@ public Object answer(InvocationOnMock invocation) {
     }
 
     @Test
-    public void getSiteID() {
-        doAPICallWithAnswer(true);
+    public void getSiteIDSuccess() {
+        String response = "{\"site_id\":\"http://google.com\"}";
+        doAPICallWithAnswer(true,response);
 
         sourcePointClient.getSiteID(onLoadComplete);
 
         verify(onLoadComplete, times(1)).onSuccess(eq("http://google.com"));
-
         verify(onLoadComplete, never()).onFailure(any(ConsentLibException.class));
-
         reset(onLoadComplete);
+    }
 
-        doAPICallWithAnswer(false);
+    @Test
+    public void getSiteIDFailure(){
+        String response = "{\"site_id\":\"http://google.com\"}";
+        doAPICallWithAnswer(false,response);
 
         sourcePointClient.getSiteID(onLoadComplete);
 
         verify(onLoadComplete, never()).onSuccess(any());
-
         verify(onLoadComplete, times(1)).onFailure(any(ConsentLibException.class));
+        reset(onLoadComplete);
```

Comment:
What if we move the line:
```java
reset(onLoadComplete);
```
to a `@After`?
```java
@After
public void resetOnLoadCompleteMock() {
  reset(onLoadComplete);
}
```
Would that work?

Reply:
yes, that would also work.

Reply:
Refactored the code